### PR TITLE
Remove the dependency on pyOpenSSL

### PR DESCRIPTION
**Pull request by nextgens**
## What type of PR?

enhancement

## What does this PR do?

Remove the dependency on pyOpenSSL

## Prerequisites
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- [ ] In case of feature or enhancement: documentation updated accordingly
- [ ] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

### CODE REVIEW

**Comment by nextgens** [comment]
bors try

File: core/admin/mailu/dkim.py [hunk 1]
```diff
@@ -2,20 +2,21 @@
...
+    k = rsa.generate_private_key(public_exponent=65537, key_size=bits)
+    return k.private_bytes(encoding=serialization.Encoding.PEM,format=serialization.PrivateFormat.PKCS8,encryption_algorithm=serialization.NoEncryption())
 
 
 def strip_key(pem):
     """ Return only the b64 part of the ASCII armored PEM.
     """
-    key = crypto.load_privatekey(crypto.FILETYPE_PEM, pem)
-    public_pem = crypto.dump_publickey(crypto.FILETYPE_PEM, key)
+
```
Comment (ghostwheel42) [inline core/admin/mailu/dkim.py:19]: please remove this newline ;)

File: core/admin/mailu/schemas.py [hunk 2]
```diff
@@ -609,7 +609,7 @@ def _deserialize(self, value, attr, data, **kwargs):
 
         # check key validity
         try:
-            crypto.load_privatekey(crypto.FILETYPE_PEM, value)
+            serialization.load_pem_private_key(value, password=None)
         except crypto.Error as exc:
             raise ValidationError(f'invalid dkim key {bad_key!r}') from exc
```
Comment (ghostwheel42) [inline core/admin/mailu/schemas.py:614]: The eception handling needs to be changed and "load" only eats bytes:
```
        try:
            serialization.load_pem_private_key(bytes(value, "ascii"), password=None)
        except (UnicodeEncodeError, ValueError) as exc:
            raise ValidationError(f'invalid dkim key {bad_key!r}') from exc
```

**Review by ghostwheel42 — CHANGES_REQUESTED** [review]
Nice It's good to remove openssl here.

File: core/admin/mailu/dkim.py (hunk 1 above)
Comment (nextgens) [reply core/admin/mailu/dkim.py]: That's [b553d02](https://github.com/Mailu/Mailu/pull/2567/commits/b553d025eb029fe4189f0c51fbe60332203540e5)

**Review by ghostwheel42 — APPROVED** [review]
LGTM

**Pull request merged**